apps/api/src/app/events/usecases/send-message/test-send-email.usecase.ts

```diff
@@ -65,25 +70,40 @@ export class SendTestEmail {
       })
     );
 
-    const mailData: IEmailOptions = {
-      to: Array.isArray(email) ? email : [email],
-      subject,
-      html: html as string,
-      from: command.payload.$sender_email || integration?.credentials.from || 'no-reply@novu.co',
-    };
-
     if (email && integration) {
-      await this.sendMessage(integration, mailData, mailFactory);
+      const mailData: IEmailOptions = {
+        to: Array.isArray(email) ? email : [email],
+        subject,
+        html: html as string,
+        from: command.payload.$sender_email || integration?.credentials.from || 'no-reply@novu.co',
+      };
+
+      await this.sendMessage(integration, mailData, mailFactory, command);
 
       return;
     }
   }
 
-  private async sendMessage(integration: IntegrationEntity, mailData: IEmailOptions, mailFactory: MailFactory) {
-    const mailHandler = mailFactory.getHandler(integration, mailData.from);
+  private async sendMessage(
+    integration: IntegrationEntity,
+    mailData: IEmailOptions,
+    mailFactory: MailFactory,
+    command: TestSendMessageCommand
+  ) {
+    const { providerId } = integration;
+    const mailHandler = mailFactory.getHandler(
+      { ...integration, providerId: GetNovuIntegration.mapProviders(ChannelTypeEnum.EMAIL, providerId) },
+      mailData.from
+    );
 
     try {
       await mailHandler.send(mailData);
+      this.analyticsService.track('Test Email Sent - [Events]', command.userId, {
+        _organization: command.organizationId,
+        _environment: command.environmentId,
+        channel: ChannelTypeEnum.EMAIL,
+        providerId,
+      });
```

Comment:
added a new segment analytics event - please note it's sent to analytics after the email is sent